### PR TITLE
49 details presentation layer integration

### PR DESCRIPTION
**Pull request by EslamHamouda**

### CODE REVIEW

File: app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+) : RecyclerView.Adapter<MerchantsAdapter.ViewHolder>() {
+    inner class ViewHolder(private val binding: ListItemMerchantBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: MerchantDomainModel) {
+            binding.tvMerchantName.text = item.arabicName
+            binding.tvMerchantAddress.text = item.address
+            binding.ivPhone.setOnClickListener {
+                val uri = "tel:${item.phoneNumber}"
+                val intent = Intent(Intent.ACTION_VIEW, Uri.parse(uri))
+                startActivity(binding.root.context, intent, null)
```
Comment (KhaledSherifSayed) [inline app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt]: you can add this logic in fragment or activity which contain adapter by listener

File: app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            binding.ivMap.setOnClickListener {
+                val uri =
```
Comment (KhaledSherifSayed) [inline app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt]: same

File: app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt (hunk 1 above)
Comment (EslamHamouda) [reply app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt]: Confirmed 🚀

File: app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt (hunk 2 above)
Comment (EslamHamouda) [reply app/src/main/java/com/android/halanvendordiscovery/presentation/details/adapter/MerchantsAdapter.kt]: Confirmed 🚀

**Review by m-taj — APPROVED** [review]

**Pull request merged**